src/NServiceBus.Persistence.DynamoDB/Installer.cs

```diff
@@ -6,73 +6,115 @@
+                {
+                    log.Error(
+                        $"Could not complete the installation. An error occurred while creating the sagas table: {installerSettings.SagaTableName}",
+                        e);
+                    throw;
+                }
             }
         }
 
         async Task CreateOutboxTableIfNotExists(IAmazonDynamoDB client, string outboxTableName,
             CancellationToken cancellationToken)
         {
-            // TODO: Maybe these should be configurable?
             var createTableRequest = new CreateTableRequest
             {
                 TableName = outboxTableName,
-                AttributeDefinitions = new List<AttributeDefinition>()
+                AttributeDefinitions = new List<AttributeDefinition>
                 {
-                    new() { AttributeName = "PK", AttributeType = "S" },
-                    new() { AttributeName = "SK", AttributeType = "S" }
+                    new() { AttributeName = outboxConfiguration.PartitionKeyName, AttributeType = ScalarAttributeType.S },
+                    new() { AttributeName = outboxConfiguration.SortKeyName, AttributeType = ScalarAttributeType.S },
                 },
-                KeySchema = new List<KeySchemaElement>()
+                KeySchema = new List<KeySchemaElement>
                 {
-                    new() { AttributeName = "PK", KeyType = "HASH" },
-                    new() { AttributeName = "SK", KeyType = "RANGE" },
+                    new() { AttributeName = outboxConfiguration.PartitionKeyName, KeyType = KeyType.HASH },
+                    new() { AttributeName = outboxConfiguration.SortKeyName, KeyType = KeyType.RANGE },
                 },
-                // TODO: Fix this, allow this to be configurable
-                BillingMode = BillingMode.PAY_PER_REQUEST
+                BillingMode = installerSettings.BillingMode,
+                ProvisionedThroughput = installerSettings.ProvisionedThroughput,
             };
 
             await CreateTable(client, createTableRequest, cancellationToken).ConfigureAwait(false);
             await WaitForTableToBeActive(client, outboxTableName, cancellationToken).ConfigureAwait(false);
+            await ConfigureTimeToLive(client, outboxTableName, cancellationToken).ConfigureAwait(false);
+
         }
 
-        async Task CreateSagaTableIfNotExists(IAmazonDynamoDB client, string sagaTableName, CancellationToken cancellationToken)
+        async Task ConfigureTimeToLive(IAmazonDynamoDB client, string outboxTableName, CancellationToken cancellationToken)
+        {
+            var ttlDescription = await client.DescribeTimeToLiveAsync(outboxTableName, cancellationToken).ConfigureAwait(false);
+
+            if (ttlDescription.TimeToLiveDescription.AttributeName != null)
+            {
+                if (ttlDescription.TimeToLiveDescription.AttributeName == outboxConfiguration.TimeToLiveAttributeName)
```

Comment:
Syncinc with Dan: This might be something we should do as sort of startup check, also when we're not creating the tables. This would allow the installation logic to dismiss this check and just swallow any exceptions.

Reply:
discussing with daniel: what permissions do we require for validation?

Reply:
We are tracking this question but will move ahead with the current implementation